cpp/include/kvikio/mmap.hpp

```diff
@@ -0,0 +1,211 @@
+   *
+   * @return Initial file offset of the mapped region
+   */
+  std::size_t initial_file_offset() const noexcept;
+
+  /**
+   * @brief Get the file size if the file is open. Returns 0 if the file is closed.
+   *
+   * The behavior of this method is consistent with `FileHandle::nbytes`.
+   *
+   * @return The file size in bytes
+   */
+  [[nodiscard]] std::size_t file_size() const;
+
+  /**
+   * @brief Alias of `file_size`
+   *
+   * @return The file size in bytes
+   */
+  [[nodiscard]] std::size_t nbytes() const;
+
+  /**
+   * @brief Whether the mapping handle is closed
+   *
+   * @return Boolean answer
+   */
+  [[nodiscard]] bool closed() const noexcept;
+
+  /**
+   * @brief Close the mapping handle if it is open; do nothing otherwise
+   */
+  void close() noexcept;
+
+  /**
+   * @brief Sequential read `size` bytes from the file (with the offset `file_offset`) to the
+   * destination buffer `buf`
+   *
+   * @param buf Address of the host or device memory (destination buffer)
+   * @param size Size in bytes to read. If not specified, read starts from `file_offset` to the end
+   * of file
+   * @param file_offset File offset
+   * @return Number of bytes that have been read
+   *
+   * @exception std::out_of_range if the read region specified by `file_offset` and `size` is
+   * outside the initial region specified when the mapping handle was constructed
+   * @exception std::invalid_argument if the size is given but is 0
+   * @exception std::runtime_error if the mapping handle is closed
+   */
+  std::size_t read(void* buf,
+                   std::optional<std::size_t> size = std::nullopt,
+                   std::size_t file_offset         = 0);
+
+  /**
+   * @brief Parallel read `size` bytes from the file (with the offset `file_offset`) to the
+   * destination buffer `buf`
+   *
+   * @param buf Address of the host or device memory (destination buffer)
+   * @param size Size in bytes to read. If not specified, read starts from `file_offset` to the end
+   * of file
+   * @param file_offset File offset
+   * @return Future that on completion returns the size of bytes that were successfully read.
+   *
+   * @exception std::out_of_range if the read region specified by `file_offset` and `size` is
+   * outside the initial region specified when the mapping handle was constructed
+   * @exception std::invalid_argument if the size is given but is 0
+   * @exception std::runtime_error if the mapping handle is closed
+   *
+   * @note The `std::future` object's `wait()` or `get()` should not be called after the lifetime of
+   * the MmapHandle object ends. Otherwise, the behavior is undefined.
+   */
+  std::future<std::size_t> pread(void* buf,
+                                 std::optional<std::size_t> size = std::nullopt,
+                                 std::size_t file_offset         = 0,
+                                 std::size_t mmap_task_size      = defaults::mmap_task_size());
```

Comment:
I think we should use the same naming as the other handles.
```suggestion
                                 std::size_t task_size      = defaults::mmap_task_size());
```

Reply:
Done. Also applied to `mmap.cpp` except for the exception message.